main/test/flix/Test.Type.Bool.flix

```diff
@@ -0,0 +1,191 @@
+
+@Test
+def test03(): Bool = assertEqual(mkFalse(), mkFalse())
+
+@Test
+def test04(): Bool = assertEqual(mkTrue(), mkTrue())
+
+@Test
+def test05(): Bool = assertUnequal(mkTrue(), mkFalse())
+
+@Test
+def test06(): Bool = assertUnequal(mkFalse(), mkTrue())
+
+@Test
+def test07(): Bool = assertFalse(mkNot(mkTrue()))
+
+@Test
+def test08(): Bool = assertTrue(mkNot(mkFalse()))
+
+@Test
+def test09(): Bool = assertTrue(mkNot(mkNot(mkTrue())))
+
+@Test
+def test10(): Bool = assertFalse(mkNot(mkNot(mkFalse())))
+
+@Test
+def test11(): Bool = assertFalse(mkNot(mkNot(mkNot(mkTrue()))))
+
+@Test
+def test12(): Bool = assertTrue(mkNot(mkNot(mkNot(mkFalse()))))
+
+//Function composition
+
+@Test
+def test13(): Bool = let id = mkNot >> mkNot; assertTrue(id(mkTrue()))
+
+@Test
+def test14(): Bool = let id = mkNot >> mkNot; assertTrue(id(mkTrue()))
+
+@Test
+def test15(): Bool = let notId = mkNot >> mkNot >> mkNot; assertFalse(notId(mkTrue()))
+
+@Test
+def test16(): Bool = let notId = mkNot >> mkNot >> mkNot; assertFalse(notId(mkTrue()))
+
+
+def doubleSwaps(index: Int32, k: B[b] -> B[not b]): B[b] -> B[not b] = {
+    if (index == 0) k else doubleSwaps(index - 1, mkNot >> mkNot >> k)
+}
+
+@Test
+def test17(): Bool = let f = doubleSwaps(0, mkNot); assertTrue(f(mkFalse()))
+
+@Test
+def test18(): Bool = let f = doubleSwaps(5, mkNot); assertTrue(f(mkFalse()))
+
+@Test
+def test19(): Bool = let f = doubleSwaps(100, mkNot); assertTrue(f(mkFalse()))
+
+@Test
+def test20(): Bool = {
+    let f = doubleSwaps(20, mkNot);
+    let g = doubleSwaps(21, mkNot);
+    assertTrue(f(g(mkTrue())))
+}
+
+//mkAnd() and mkOr() cases
+
+@Test
+def test21(): Bool = assertFalse(mkAnd(mkFalse(), mkFalse()))
+
+@Test
+def test22(): Bool = assertFalse(mkAnd(mkFalse(), mkTrue()))
+
+@Test
+def test23(): Bool = assertFalse(mkAnd(mkTrue(), mkFalse()))
+
+@Test
+def test24(): Bool = assertTrue(mkAnd(mkTrue(), mkTrue()))
+
+@Test
+def test25(): Bool = assertFalse(mkOr(mkFalse(), mkFalse()))
+
+@Test
+def test26(): Bool = assertTrue(mkOr(mkTrue(), mkFalse()))
+
+@Test
+def test27(): Bool = assertTrue(mkOr(mkFalse(), mkTrue()))
+
+@Test
+def test28(): Bool = assertTrue(mkOr(mkFalse(), mkTrue()))
+
+
+//Nested mkAnd(), mkOr(), mkNot() cases
+@Test
+def test29(): Bool = assertTrue(mkNot(mkAnd(mkTrue(), mkFalse())))
+
+@Test
+def test30(): Bool = assertFalse(mkNot(mkAnd(mkTrue(), mkTrue())))
+
+@Test
+def test31(): Bool = assertTrue(mkNot(mkOr(mkFalse(), mkFalse())))
+
+@Test
+def test32(): Bool = assertFalse(mkNot(mkOr(mkFalse(), mkTrue())))
+
+@Test
+def test33(): Bool = assertEqual(mkOr(mkFalse(), mkFalse()), mkAnd(mkTrue(), mkFalse()))
+
+@Test
+def test34(): Bool = assertEqual(mkOr(mkFalse(), mkTrue()), mkAnd(mkTrue(), mkTrue()))
+
+@Test
+def test35(): Bool = assertEqual(mkNot(mkAnd(mkTrue(), mkFalse())), mkNot(mkOr(mkFalse(), mkFalse())))
+
+@Test
+def test36(): Bool = assertTrue(mkAnd(mkAnd(mkTrue(), mkTrue()), mkOr(mkFalse(), mkNot(mkNot(mkTrue())))))
+
+
+//Operations on lists
```

Comment:
I see what you did, but there is no need. The list aspect does not add anything because the type of the elements must be the same, so I would delete these tests.